src/components/Form.js

```diff
@@ -0,0 +1,182 @@
+import { useForm } from 'react-hook-form'
+import PropTypes from 'prop-types'
+import InputField from './InputField'
+import Button from './Button'
+import styled from 'styled-components'
+import { useState, useRef } from 'react'
+import { yupResolver } from '@hookform/resolvers/yup'
+import * as yup from 'yup'
+import React from 'react'
+
+Form.propTypes = {
+  onSubmit: PropTypes.func,
+}
+
+const schema = yup.object().shape({
+  name: yup.string().required(),
+  phone: yup.number().positive().integer().required(),
+  email: yup.string().required(),
+  department: yup.string().required(),
+  skill1: yup.string().required(),
+  skill2: yup.string(),
+  skill3: yup.string(),
+  skill4: yup.string(),
+  skill5: yup.string(),
+  skill6: yup.string(),
```

Comment:
I think this should maybe be an array with strings so that it's not limited to 6 skills. I commented on your submit function how to do it.